sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CollationTypeCasts.scala

```diff
@@ -54,7 +52,8 @@ object CollationTypeCasts extends TypeCoercionRule {
 
     case otherExpr @ (
       _: In | _: InSubquery | _: CreateArray | _: ArrayJoin | _: Concat | _: Greatest | _: Least |
-      _: Coalesce | _: BinaryExpression | _: ConcatWs) =>
+      _: Coalesce | _: BinaryExpression | _: ConcatWs | _: StringTrim | _: StringTrimBoth |
+      _: StringTrimLeft | _: StringTrimRight) =>
```

Comment:
StringTrimBoth is replaced with StringTrim, so it is unnecessary to add it to rules here.